packages/governance/src/types.js

```diff
@@ -87,74 +151,101 @@
+ * @property {() => Question} getQuestion
+ * @property {() => Promise<Position>} getOutcome
+ * @property {() => QuestionDetails} getDetails
  * @property {() => Promise<VoteStatistics>} getStats
  */
 
 /**
- * @typedef {Object} BallotCounterCloseFacet
+ * @typedef {Object} VoteCounterCloseFacet
  *   TEST ONLY: Should not be allowed to escape from contracts
  * @property {() => void} closeVoting
  */
 
 /**
- * @typedef {Object} CompleteEqualWeightBallot
- * @property {string} question
- * @property {string[]} chosen - a list of equal-weight preferred positions
+ * @typedef {Object} VoteCounterFacets
+ * @property {VoteCounterPublicFacet} publicFacet
+ * @property {VoteCounterCreatorFacet} creatorFacet
+ * @property {VoteCounterCloseFacet} closeFacet
  */
 
 /**
- * @typedef {Object} CompleteWeightedBallot
- * @property {string} question
- * @property {Record<string,bigint>[]} weighted - list of positions with weights.
- *   BallotCounter may limit weights to a range or require uniqueness.
+ * @callback BuildVoteCounter
+ * @param {QuestionSpec} questionSpec
+ * @param {bigint} threshold - questionSpec includes quorumRule; the registrar
+ *    converts that to a number that the counter can enforce.
+ * @param {Instance} instance
+ * @returns {VoteCounterFacets}
  */
 
 /**
- * @typedef {Object} CompleteOrderedBallot
- * @property {string} question
- * @property {string[]} ordered - ordered list of position from most preferred to
- *   least preferred
+ * @callback LooksLikeQuestionSpec
+ * @param {QuestionSpec} allegedQuestionSpec
+ * @returns {QuestionSpec}
  */
 
 /**
- * @typedef { CompleteEqualWeightBallot | CompleteOrderedBallot | CompleteWeightedBallot } CompletedBallot
+ * @callback LooksLikeIssueForType
+ * @param {ElectionType} electionType
+ * @param {Issue} issue
  */
 
 /**
  * @callback SubmitVote
- * @param {Handle<'Voter'>} seat
- * @param {ERef<CompletedBallot>} filledBallot
+ * @param {Handle<'Voter'>} voterHandle
+ * @param {[Position]} chosenPositions
  * @param {bigint=} weight
  */
 
 /**
- * @typedef {Object} VoterFacet
- * @property {SubmitVote} submitVote
+ * @typedef {Object} RegistrarPublic
+ * @property {() => Subscription<QuestionDetails>} getQuestionSubscription
+ * @property {() => ERef<Handle<'Question'>[]>} getOpenQuestions,
+ * @property {() => string} getName
+ * @property {() => Instance} getInstance
+ * @property {(h: Handle<'Question'>) => ERef<Question>} getQuestion
+ */
+
+/**
+ * @typedef {Object} PoserFacet
+ * @property {AddQuestion} addQuestion
+ */
+
+/**
+ * @typedef {Object} RegistrarCreatorFacet
+ *  addQuestion() can be used directly when the creator doesn't need any
+ *  reassurance. When someone needs to connect addQuestion to the Registrar
+ *  instance, getPoserInvitation() lets them get addQuestion with assurance.
+ * @property {() => ERef<Invitation>} getPoserInvitation
+ * @property {AddQuestion} addQuestion
+ * @property {() => Invitation[]} getVoterInvitations
```

Comment:
This conflicts with the usage of `zcf.makeInvitation`. So...

```suggestion
 * @property {() => Promise<Invitation>[]} getVoterInvitations
```

Reply:
done